pkg/config/config.go

```diff
@@ -41,6 +41,20 @@ func (r Repo) String() string {
 	return r.Owner + "/" + r.Name
 }
 
+// GiteaRepo represents repo hosted on Gitea instance
+type GiteaRepo struct {
```

Comment:
Fine for me atm, because I guess we will unify the repo in another PR to one  *repo* struct

Reply:
We could, but then the users would have to know how to construct the API URLs for their instance and then in the client creation code we would have to parse that URL to pass to 
https://godoc.org/code.gitea.io/sdk/gitea#NewClient

Reply:
Well for public github/gitlab we defined the default urls but if a user host its enterprise/private one, then he/she should know how to build the full url. That's why we have the `api` and `download` fields. 

Imagine they host their vcs behind a LoadBalancer or Proxy which expects the api at `/foo/bar` and rewrites to `/api/v1/` where the api acually ist. We don't know the user's configuration, so we give them their freedom to completely define it

Reply:
So you propose to parse the API URL, say `https://gitea.dtluna.net/api/v1` and construct `https://gitea.dtluna.net` for [gitea.NewClient](https://godoc.org/code.gitea.io/sdk/gitea#NewClient) manually in the code?

Reply:
Yes. A good point for the validation could be parsing the url. ATM we only do it at `client/github.go`. What do you think @caarlos0 ?

Reply:
sounds good @mavogel :)